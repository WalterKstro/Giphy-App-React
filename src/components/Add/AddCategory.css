input[type=text]{
    outline: none;
    width:95%;
    display:block;
    margin-left:auto;
    margin-right: auto;
    height:4rem;
    padding:1rem 2rem;
    border:1px solid #20232a;
    border-radius: .5rem;
}

.form_add{
    margin-bottom: 2rem;
}


@media screen and (min-width:600px) {
    input[type=text] {
        width:80%;
    }
}

@media screen and (min-width:768px) {
    input[type=text] {
        width:60%;
    }
}

@media screen and (min-width:1024px) {
    input[type=text] {
        width:40%;
    }
}