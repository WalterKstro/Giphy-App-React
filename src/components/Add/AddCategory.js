import { useState } from 'react'
import PropTypes from 'prop-types'

import './AddCategory.css'

const AddCategory = ({setCategories}) => {
    const [valueInput, setValueInput] = useState('')

    const handleChange = event => {
        setValueInput( event.target.value )
    }

    const handleSubmit = event => {
        event.preventDefault()
        if(valueInput.trim() !== '') {
            setCategories( state => [valueInput,...state] )
            setValueInput('')
        }
    }
    
    return (
        <form onSubmit={handleSubmit} className="form_add">
            <input 
                type="text"
                placeholder="Add new category"
                value={valueInput}
                onChange={ handleChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
}

export default AddCategory
