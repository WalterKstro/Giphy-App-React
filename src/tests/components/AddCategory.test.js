import { shallow } from "enzyme"
import AddCategory from "../../components/Add/AddCategory"

describe('Test of the component <AddCategory/>', () => {
    const setCategories = jest.fn()
    let component

    beforeEach(() => {
        jest.clearAllMocks()
        component = shallow(<AddCategory setCategories={setCategories} />)
    })

    test('should show a snapshot of component <AddCategory/>', () => {
        expect(component).toMatchSnapshot()
    })

    test('should change the value of input', () => {
        const value = 'Hola Mundo con Enzyme'
        const input = component.find('input').simulate('change', {target: {value}})
        const testValue = component.find('p').text()

        expect(testValue).toBe(value)

    })

    test('no should throw the event submit, if input value is empty', () => {
        const form = component.find('form').simulate('submit', {preventDefault : jest.fn()})

        expect(setCategories).not.toHaveBeenCalled()

    })
    


    test('should call the function setCategories and clear the input', () => {
        const value = 'Hola Mundo'

        const inputChange = component.find('input')
              inputChange.simulate('change', { target: {value}})
        const mockSubmit = component.find('form')
              mockSubmit.simulate('submit',{ preventDefault: jest.fn()})

        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) )

        expect(inputChange.text()).toBe('')
        
    })
    
    
        
})