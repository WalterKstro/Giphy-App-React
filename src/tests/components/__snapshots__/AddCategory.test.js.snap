// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of the component <AddCategory/> should show a snapshot of component <AddCategory/> 1`] = `
<form
  className="form_add"
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Add new category"
    type="text"
    value=""
  />
</form>
`;
